upstairs/src/lib.rs

```diff
@@ -4769,7 +4792,23 @@ impl Downstairs {
             }
             // Now that we've collected jobs to retire, remove them from the map
             for &id in &retired {
-                let _ = self.ds_active.remove(&id);
+                let job = self.ds_active.remove(&id);
+                // Update pending bytes when this job is retired
+                match &job.work {
+                    IOop::Write { writes, .. }
+                    | IOop::WriteUnwritten { writes, .. } => {
+                        self.write_bytes_outstanding = self
+                            .write_bytes_outstanding
+                            .checked_sub(
+                                writes
+                                    .iter()
+                                    .map(|w| w.data.len() as u64)
+                                    .sum::<u64>(),
+                            )
+                            .unwrap();
```

Comment:
When would we hit this unwrap?  If our accounting was wrong?

Reply:
Yup, exactly.  This already caught one bug in development, where I forgot to add bytes from skipped jobs (but still subtracted them here, causing an overflow).